doc/keynames/README.md

```diff
@@ -0,0 +1,477 @@
+# Key Names in Elektra
+
+## Preface
+
+This document is a full explanation of how Key Names work in Elektra.
+In addition to this document, a reference Python implementation can be found in [keynames.py](keynames.py).
+The goal this Python implementation is not to be fast, or to be used in any way other than as a reference.
+If there are any discrepancies between this document, the Python implementation and the actual C implementation in [src/libs/elektra/keyname.c](../../src/libs/elektra/keyname.c), you should consider them as follows:
+
+1. The C implementation is optimized for speed and much harder to maintain.
+   It is mostly likely to be incorrect.
+2. In most cases, this document outranks the Python implementation.
+   There may, however, be cases where the language in this document was too vague and the Python implementation is actually correct.
+3. If two of the sources agree, the third one is probably incorrect.
+   Although again, if one of the agreeing sources is the C implementation it could still be the case that there is a mistake.
+
+In any case: If you find a discrepancy, please file a bug report at https://issues.libelektra.org/new.
+
+> _Note:_ Mistakes happen.
```

Comment:
The note can be removed. It is enough if people know what our prioritization is.